packages/api-plugin-promotions-discounts/src/discountTypes/shipping/applyShippingDiscountToCart.js

```diff
@@ -1,5 +1,117 @@
 /* eslint-disable no-unused-vars */
-import ReactionError from "@reactioncommerce/reaction-error";
+import { createRequire } from "module";
+import _ from "lodash";
+import Logger from "@reactioncommerce/logger";
+import recalculateShippingDiscount from "../../utils/recalculateShippingDiscount.js";
+import getTotalDiscountOnCart from "../../utils/getTotalDiscountOnCart.js";
+import formatMoney from "../../utils/formatMoney.js";
+import getEligibleShipping from "../../utils/getEligibleIShipping.js";
+
+const require = createRequire(import.meta.url);
+
+const pkg = require("../../../package.json");
+
+const { name, version } = pkg;
+const logCtx = {
+  name,
+  version,
+  file: "shipping/applyShippingDiscountToCart.js"
+};
+
+/**
+ * @summary Map discount record to shipping discount
+ * @param {Object} params - The action parameters
+ * @param {Object} discountedItem - The item that were discounted
+ * @returns {Object} Shipping discount record
+ */
+export function createDiscountRecord(params, discountedItem) {
+  const { promotion, actionParameters } = params;
+  const shippingDiscount = {
+    promotionId: promotion._id,
+    discountType: actionParameters.discountType,
+    discountCalculationType: actionParameters.discountCalculationType,
+    discountValue: actionParameters.discountValue,
+    discountMaxValue: actionParameters.discountMaxValue,
+    dateApplied: new Date(),
+    discountedItemType: "shipping",
+    discountedAmount: discountedItem.amount,
+    stackability: promotion.stackability,
```

Comment:
Are we applying the shipping discount to items? Not sure that's how we want to do that